tox.ini

```diff
@@ -18,41 +23,57 @@ commands = pre-commit run --all-files --show-diff-on-failure --color=always
 [testenv:py{,38,39,310,311,312}{,-fast}]
 depends: pre-commit
 use_develop = fast: True
-skip_sdist = fast: True
 # Needed on Windows to test data directory creation
 passenv = LOCALAPPDATA
+setenv = COVERAGE_FILE = {env:COVERAGE_FILE:.coverage}
 extras = dev
 # 2023-04-22 see pkgenv ↑
 download = {[pkgenv]download}
 commands =
-    !fast : python -m coverage run -m pytest {posargs:-vv}
-    fast : python -m pytest {posargs:-vv -n auto}
+    !fast : python -m coverage run -m pytest {posargs:-vv --color yes}
+    fast : python -m pytest {posargs:-vv --color yes -n auto}
 
-[testenv:coverage{,-html}{,-fail}]
+[testenv:coverage{,38,39,310,311,312}{,-ci}{,-platform,-platform-linux,-platform-macos,-platform-windows,-project}{,-keep}{,-html}]
 depends = py{,38,39,310,311,312}
-# coverage should run on oldest supported Python
-base_python = py38,py39,py310,py311
-parallel_show_output = True
+# by default, coverage should run on oldest supported Python for testing platform coverage.
+# however, coverage for a particular Python version should match the version used for pytest.
+base_python =
+    coverage: py38,py39,py310,py311,py312
+    coverage38: py38
+    coverage39: py39
+    coverage310: py310
+    coverage311: py311
+    coverage312: py312
 extras = dev
 # 2023-04-22 see pkgenv ↑
 download = {[pkgenv]download}
+passenv = COVERAGE_FILE
+setenv =
+    keep: COMBINE_FLAGS = --keep
+    # spoof platform for conditional coverage exclusions
+    platform-linux: COVERAGE_PLATFORM = linux
+    platform-macos: COVERAGE_PLATFORM = darwin
+    platform-windows: COVERAGE_PLATFORM = win32
+    # use the coverage files created in CI
+    ci-platform-linux: COVERAGE_FILE = .coverage.ubuntu
+    ci-platform-macos: COVERAGE_FILE = .coverage.macos
+    ci-platform-windows: COVERAGE_FILE = .coverage.windows
+    # disable conditional coverage exclusions for Python version to test entire platform
+    {platform,project}: COVERAGE_EXCLUDE_PYTHON_VERSION=disable
+    # disable conditional coverage exclusions for host platform to test entire project
+    project: COVERAGE_EXCLUDE_PLATFORM=disable
```

Comment:
I keep forgetting that `tox`'s `factor`s are intended to control how the environment runs....so, I converted the `COVERAGE_*` env vars to `factor`s. I'm not sure if I've sold myself on the `ci` factor yet....maybe there's a better option...